File: 2262-total-appeal-of-a-string/2262-total-appeal-of-a-string.py
class Solution:
    def appealSum(self, s: str) -> int:
        res, curr, d = 0, 0, defaultdict(lambda: -1)
        for i, val in enumerate(s):
            curr += (i - d[val])
            res += curr
            d[val] = i
        return res
        

File: 2262-total-appeal-of-a-string/2262-total-appeal-of-a-string.cpp
class Solution {
public:
    long long appealSum(string s) {
        long long res = 0;
        long long curr = 0;
        unordered_map<int, int> m;
        for(int i = 0; i < s.size(); ++i) {
            if(m.find(s[i]) == m.end()) {
                curr += (i + 1);
            } else {
                curr += (i - m[s[i]]);
            }
            m[s[i]] = i;
            res += curr;
        }
        return res;
    }
};